### ISSUE TITLE
[Bug]: [Accessibility] [Divider] Not visible in high contrast

### ISSUE THREAD

**Issue by snisonoff-microsoft-unmanaged** [microsoft/fluentui#31214]
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
Repro steps:
- Open code sandbox
- open windows settings, go to Accessibility, Contrast themes
- Test the Divider in the code sandbox against Aquatic, Desert, Dusk, and Night Sky themes
```

### Are you reporting Accessibility issue?

None

### Reproduction

https://codesandbox.io/p/sandbox/skeleton-forked-86qstz?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clvhcjtj700063j6i4qnn80hj%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clvhcjtj600023j6i8cpm2blj%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clvhcjtj600033j6il37sevlt%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clvhcjtj600053j6ier5vl1py%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clvhcjtj600023j6i8cpm2blj%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clvhcjtj600013j6i2n9952mz%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Findex.tsx%2522%257D%255D%252C%2522id%2522%253A%2522clvhcjtj600023j6i8cpm2blj%2522%252C%2522activeTabId%2522%253A%2522clvhcjtj600013j6i2n9952mz%2522%257D%252C%2522clvhcjtj600053j6ier5vl1py%2522%253A%257B%2522id%2522%253A%2522clvhcjtj600053j6ier5vl1py%2522%252C%2522activeTabId%2522%253A%2522clvhcm2qf003j3j6idciq3tq4%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clvhcjtj600043j6i1bs6qwue%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%252C%257B%2522type%2522%253A%2522SANDBOX_INFO%2522%252C%2522id%2522%253A%2522clvhcm2qf003j3j6idciq3tq4%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%257D%252C%2522clvhcjtj600033j6il37sevlt%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clvhcjtj600033j6il37sevlt%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D

## Actual Behavior
Divider is not clearly visible in any of the high contrast themes except for Desert

## Expected Behavior
Divider should be clearly visible in all of the high contrast themes
Aquatic:
![image](https://github.com/microsoft/fluentui/assets/167139608/7b770f55-3fc4-4e89-b900-a614e3606f31)
![image](https://github.com/microsoft/fluentui/assets/167139608/50febedc-b333-4467-89aa-4d4be8b911fe)
![image](https://github.com/microsoft/fluentui/assets/167139608/5285d66a-6ee3-4a46-93d3-0f8cc70c5160)

### Requested priority

Blocking

### Products/sites affected

TwoUI

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by smhigley** [comment]
We investigated, and it turns out this is an Edge bug where `border-color` is not picking up the correct high contrast theme token when defined on pseudo-elements.

We've logged a bug on Edge, and added an entry to our [internal wiki](https://dev.azure.com/microsoftdesign/fluent-ui/_wiki/wikis/fluent-ui.wiki/334/Fluent-v9-known-issues?anchor=divider-is-not-clearly-visible-in-high-contrast-mode) (where you can find the browser bug link as well).

Thanks for finding and filing this! I'm going to close the github issue, since it's external and being tracked in Edge.
